pkg/storage/unified/apistore/prepare.go

```diff
@@ -71,9 +77,20 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 	if err != nil {
 		return nil, err
 	}
+
+	if previous.GetUID() == "" {
+		return nil, fmt.Errorf("object is missing UID")
+	}
+
 	obj.SetUID(previous.GetUID())
 	obj.SetCreatedBy(previous.GetCreatedBy())
 	obj.SetCreationTimestamp(previous.GetCreationTimestamp())
+	obj.SetResourceVersion("")
+	obj.SetName(previous.GetName())
+
+	if obj.GetGenerateName() == "" {
+		obj.SetGenerateName(obj.GetName())
```

Comment:
i'm not following why this line would be needed. generateName drives the creation of a name if name is empty

Reply:
correct, I didn't see we were already validating this, removing 👍

Reply:
This needs to be removed.  if generate name exists here, something already went wrong